apps/wallet/src/ui/app/shared/InputWithAction.tsx

```diff
@@ -15,37 +16,68 @@ export interface InputWithActionProps
     onActionClicked?: PillProps['onClick'];
     actionType?: PillProps['type'];
     name: string;
+    prefix?: string;
```

Comment:
these props are used only when the Input is NumberInput and might be a bit confusing. I wonder if adding an `as` prop to be an `input` by default and derive the props from that would be a bit more clear.